File: Python/1282_group-the-people-given-the-group-size-they-belong-to.py
class Solution:
    def groupThePeople(self, groupSizes: List[int]) -> List[List[int]]:
        group_size_map = {}
        output = []

        for idx, size in enumerate(groupSizes):
            if size not in group_size_map:
                group_size_map[size] = []

            group_size_map[size].append(idx)

            if len(group_size_map[size]) == size:
                output.append(group_size_map[size][:])
                group_size_map[size].clear()

        return output

File: Elixir/1282_group-the-people-given-the-group-size-they-belong-to.ex
defmodule Solution do
  @spec group_the_people(group_sizes :: [integer]) :: [[integer]]
  def group_the_people(group_sizes) do
    Enum.with_index(group_sizes)
    |> Enum.reduce(Map.new(), fn {size, idx}, acc ->
      Map.update(acc, size, [idx], fn group -> [idx | group] end)
    end)
    |> Enum.flat_map(fn {size, group} ->
      Enum.chunk_every(group, size)
    end)
  end
end
